### PR TITLE
test agent connection during agent config.

### PR DESCRIPTION
**Pull request by TingluoHuang**

### CODE REVIEW

File: src/Agent.Listener/Configuration/ConfigurationManager.cs [hunk 1]
```diff
@@ -277,6 +275,14 @@ public AgentSettings LoadSettings()
                 _store.SaveCredential(credProvider.CredentialData);
             }
 
+            // Testing agent connection, detect any protential connection issue, like local clock skew that cause OAuth token expired.
+            _term.WriteLine(StringUtil.Loc("TestAgentConnection"));
+            var credMgr = HostContext.GetService<ICredentialManager>();
+            VssCredentials credential = credMgr.LoadCredentials();
+            VssConnection conn = ApiUtil.CreateConnection(new Uri(serverUrl), credential);
+            var agentSvr = HostContext.GetService<IAgentServer>();
+            await agentSvr.ConnectAsync(conn);
```
Comment (TingluoHuang) [inline src/Agent.Listener/Configuration/ConfigurationManager.cs]: Catch TokenRequestException, print error suggestion.

Comment (bryanmacfarlane) [inline src/Agent.Listener/Configuration/ConfigurationManager.cs]: Yes, make sure it's actionable and suggests clock skew

File: src/Agent.Listener/MessageListener.cs [hunk 2]
```diff
@@ -310,6 +322,28 @@ private bool IsSessionCreationExceptionRetriable(Exception ex)
...
+            {
+                Trace.Info("Local clock might skewed.");
+                _term.WriteError(StringUtil.Loc("LocalClockSkewed"));
+                if (_sessionCreationExceptionTracker.ContainsKey(nameof(VssOAuthTokenRequestException)))
+                {
+                    _sessionCreationExceptionTracker[nameof(VssOAuthTokenRequestException)]++;
+                    if (_sessionCreationExceptionTracker[nameof(VssOAuthTokenRequestException)] * _sessionCreationRetryInterval.TotalSeconds >= _clockSkewRetryLimit.TotalSeconds)
+                    {
+                        Trace.Info("The OAuth token request exception have reached retry limit.");
+                        _term.WriteError(StringUtil.Loc("LockSkewStopRetry", _clockSkewRetryLimit.TotalSeconds));
```
Comment (ericsciple) [inline src/Agent.Listener/MessageListener.cs]: clock

File: src/Misc/layoutbin/en-US/strings.json [hunk 3]
```diff
@@ -197,6 +198,8 @@
     "bin/Agent.Listener configure --url http://mylocaltfsserver:8080/tfs --agent myagent --pool default --nostart --acceptteeeula --auth negotiate --username DEV-BOX\\developer --password MyPassword"
   ],
   "ListenForJobs": "{0:u}: Listening for Jobs",
+  "LocalClockSkewed": "Lock clock might skew too great, please sync with internet time server, time skew must be less than 5 minutes.",
+  "LockSkewStopRetry": "Stop retry on VssOAuthTokenRequestException after retried for {0} seconds.",
```
Comment (ericsciple) [inline src/Misc/layoutbin/en-US/strings.json]: The local machine's clock may be out of sync with the server time by more than five minutes. Please sync your clock with your domain or internet time and try again.

Stopped retrying OAuth token request exception after {0} seconds.

**Pull request merged**